src/Surfnet/StepupMiddleware/ApiBundle/Configuration/Service/AllowedSecondFactorListService.php

```diff
@@ -60,4 +61,16 @@ public function getAllowedSecondFactorMap()
     {
         return AllowedSecondFactorMap::from($this->allowedSecondFactorRepository->findAll());
     }
+
+    /**
+     * @param SecondFactorType $secondFactor
+     * @param Institution $institution
+     * @return bool
+     */
+    public function isSecondFactorAllowedFor(SecondFactorType $secondFactor, Institution $institution)
```

Comment:
You are correct. It's a force of habit that I want to hide everything behind a service call, but we have an intelligent (domain) object for this. Let's use that one instead.